lint/parenthesis-checker.go

```diff
@@ -27,21 +27,35 @@ func NewParenthesisChecker(ctx *Context) *ParenthesisChecker {
 // and offsers the way how to do it.
 func (c *ParenthesisChecker) Check(f *ast.File) []Warning {
 	c.warnings = c.warnings[:0]
-	for _, decl := range collectFuncDecls(f) {
-		if decl.Type.Results == nil {
-			continue
-		}
-		for _, res := range decl.Type.Results.List {
-			c.validateResultDecl(res)
+
+	for _, decl := range f.Decls {
+		switch decl := decl.(type) {
+		case *ast.FuncDecl:
+			if decl.Type.Results == nil {
+				continue
+			}
+			for _, res := range decl.Type.Results.List {
+				c.validateExpr(res.Type)
+			}
+		case *ast.GenDecl:
+			for _, spec := range decl.Specs {
+				spec, ok := spec.(*ast.ValueSpec)
+				if !ok {
+					continue
+				}
+				c.validateExpr(spec.Type)
+			}
 		}
 	}
 	return c.warnings
 }
 
-func (c *ParenthesisChecker) validateResultDecl(f *ast.Field) {
+func (c *ParenthesisChecker) validateExpr(n ast.Node) {
 	// TODO improve suggestions for complex cases like (func([](func())))
+	// TODO improve linter output to write full type, not just place
+	// whereit could be simplified
```

Comment:
`s/whereit/where it/`

Reply:
We may want to disable `interfacer` in megacheck inside our CI script.
Method says `validateExpr`, but it accepts any `ast.Node`.
And that is not quite good, because `ast.Inspect` inside it will never handle anything except expressions.

This is not the first time `interfacer` bites us.
We should treat it as a suggestion, not obligation that breaks out build.

Reply:
What do you think about `s/validateExpr/validateType/`?
Types are expressions in AST terms, but not all expressions are types.

Reply:
SGTM, i ll rename it in a moment.
CI script is out of scope of that pr. I ll create issue for that.